freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/WorkloadCredentialService.java

```diff
@@ -23,12 +27,9 @@
     public void setWorkloadCredential(FreeIpaClient freeIpaClient, String username, WorkloadCredential workloadCredential)
             throws IOException, FreeIpaClientException {
         LOGGER.debug("Setting workload credentials for user '{}'", username);
-
         try {
-            String ansEncodedKrbPrincipalKey = KrbKeySetEncoder.getASNEncodedKrbPrincipalKey(workloadCredential.getKeys());
-            freeIpaClient.userSetWorkloadCredentials(username,
-                    workloadCredential.getHashedPassword(), ansEncodedKrbPrincipalKey, workloadCredential.getExpirationDate(),
-                    workloadCredential.getSshPublicKeys().stream().map(UserManagementProto.SshPublicKey::getPublicKey).collect(Collectors.toList()));
+            String expiration = freeIpaClient.formatDate(workloadCredential.getExpirationDate());
```

Comment:
you can move the date formatting to `getOperation`. This will remove the repeated code on line 49 and simplify the method signature for `getOperation`.